### PR TITLE
Update git-tag format in generate-deployments-for-service

### PR DESCRIPTION
**Pull request by dpopacristobal**
Initially we wanted the git-tag format to be:
`paasta-{deploy_group}-{image_version}-{timestamp}-deploy`

However, this made it hard to parse when the deploy group included a `"-"`. As a result, we're instead opting for:
`paasta-{deploy_group}:{image_version}-{timestamp}-deploy`

This changes the format in the most recent changes. We'll use this format going forward in the upcoming changes.

### CODE REVIEW

**Review by jfongatyelp — COMMENTED** [review]
Apologies, we reviewed the git tag spec and `:` is not an allowed character either. 
Instead of `:`, can we use `+` ti separate `deploy_group` and `image_version`?

**Review by jfongatyelp — APPROVED** [review]

**Review by EvanKrall — APPROVED** [review]
